django/db/backends/base/validation.py

```diff
@@ -7,4 +7,17 @@ def check(self, **kwargs):
         return []
 
     def check_field(self, field, **kwargs):
-        return []
+        errors = []
+        # Ignore any related fields.
+        if hasattr(self, 'check_field_type') and not getattr(field, 'remote_field', None):
+            # Ignore fields with unsupported features.
+            db_supports_all_required_features = all(
+                getattr(self.connection.features, feature, False)
+                for feature in field.model._meta.required_db_features
+            )
+            if db_supports_all_required_features:
+                field_type = field.db_type(self.connection)
+                # Ignore non-concrete fields.
+                if field_type is not None:
+                    errors.extend(self.check_field_type(field, field_type))
+        return errors
```

Comment:
As an alternative we can write:
```python
def check_field(self, field, **kwargs):
    errors = []
    if (hasattr(self, 'check_field_type') and
            # Ignore any related fields.
            not getattr(field, 'remote_field', None) and
            # Ignore fields with unsupported features.
            all(
                getattr(self.connection.features, feature, False)
                for feature in field.model._meta.required_db_features)):
        field_type = field.db_type(self.connection)
        # Ignore non-concrete fields.
        if field_type is not None:
            errors.extend(self.check_field_type(field, field_type))
    return errors
```
I don't have a strong opinion which approach is better.